TrafficCapture/trafficReplayer/src/test/java/org/opensearch/migrations/transform/TypeMappingsExcisionTest.java

```diff
@@ -0,0 +1,72 @@
+package org.opensearch.migrations.transform;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.CharStreams;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.opensearch.migrations.replay.datahandlers.JsonAccumulator;
+import org.opensearch.migrations.replay.datahandlers.PayloadFaultMap;
+import org.opensearch.migrations.replay.datahandlers.http.HttpJsonMessageWithFaultablePayload;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedHashMap;
+
+public class TypeMappingsExcisionTest {
+
+    static final TypeReference<LinkedHashMap<String, Object>> TYPE_REFERENCE_FOR_MAP_TYPE = new TypeReference<>(){};
+
+    static ObjectMapper objectMapper = new ObjectMapper();
+
+
+    static InputStream getTypeMappingResourceStream(String resourceName) {
+        return TypeMappingsExcisionTest.class.getResourceAsStream("/sampleJsonDocuments/typeMappings/" +
+                resourceName);
+    }
+
+    @Test
+    public void removesTypeMappingsFrom_indexCreation() throws Exception {
+        var json = parseJsonFromResourceName("put_index_input.txt");
+        transformAndVerifyResult(json, "put_index_output.txt");
+    }
+
+    @Test
+    public void removesTypeMappingsFrom_documentPut() throws Exception {
+        var json = parseJsonFromResourceName("put_document_input.txt");
+        transformAndVerifyResult(json, "put_document_output.txt");
+    }
+
+    @Test
+    public void removesTypeMappingsFrom_queryGet() throws Exception {
+        var json = parseJsonFromResourceName("get_query_input.txt");
+        transformAndVerifyResult(json, "get_query_output.txt");
+    }
+
+    private static Object parseJsonFromResourceName(String resourceName) throws Exception {
+        var jsonAccumulator = new JsonAccumulator();
+        try (var resourceStream = getTypeMappingResourceStream(resourceName);
+             var isr = new InputStreamReader(resourceStream, StandardCharsets.UTF_8)) {
+            var expectedBytes = CharStreams.toString(isr).getBytes(StandardCharsets.UTF_8);
+            return jsonAccumulator.consumeByteBuffer(ByteBuffer.wrap(expectedBytes));
+        }
+    }
+
+    private static void transformAndVerifyResult(Object json, String expectedValueSource) throws Exception {
+        var jsonTransformer = getJsonTransformer();
+        json = jsonTransformer.transformJson(json);
+        var jsonAsStr = objectMapper.writeValueAsString(json);
+        Object expectedObject = parseJsonFromResourceName(expectedValueSource);
+        var expectedValue = objectMapper.writeValueAsString(expectedObject);
+        Assertions.assertEquals(expectedValue, jsonAsStr);
```

Comment:
Wouldn't this trigger false negatives if the ordering in the JSON changes? How about using either JSON Patch or Jackson to compare the JSON objects themselves instead of the serialized string versions?